TypeScript compilation error: "Generic type 'Array<T>' requires 1 type argument(s)"

Since version 9.3, the TypeScript compiler (in my case the latest 2.4.x) throws two errors regarding a missing generic for Arrays. In particular:

**First error:**
```
node_modules/github/lib/index.d.ts(1463,24): error TS2314: Generic type 'Array<T>' requires 1 type argument(s).
```
Here the line:
https://github.com/mikedeboer/node-github/blob/f34f59e2ea8d0c03c0a449e1c9f65de2fe2bb413/lib/index.d.ts#L1463

**Second error:**
```
node_modules/github/lib/index.d.ts(1471,24): error TS2314: Generic type 'Array<T>' requires 1 type argument(s).
```
Here the line:
https://github.com/mikedeboer/node-github/blob/f34f59e2ea8d0c03c0a449e1c9f65de2fe2bb413/lib/index.d.ts#L1471

**Solution:**

I assume using ```string[]``` or ```Array<string>``` in both cases should fix the issue.

--- a/lib/generateTypes.js
+++ b/lib/generateTypes.js
@@ -41,10 +41,12 @@ function replaceArrayTypes(type, name) {
         case "required_contexts":
         case "maintainers":
         case "reviewers":
+        case "team_reviewers":
         case "comments":
         case "labels":
         case "teams":
         case "users":
+        case "names":
         case "emails":
             if (type === "Array") {
                 return "string[]";
@@ -159,7 +161,7 @@ var main = module.exports = function(languageName, templateFile, outputFile) {
             methods: methods,
         });
     }, []);
- 
+
     var body = Mustache.render(template, {
         requestHeaders: requestHeaders.map(JSON.stringify),
         params: params,

--- a/lib/index.d.ts
+++ b/lib/index.d.ts
@@ -15,7 +15,7 @@ declare namespace Github {
 
   export interface EmptyParams {
   }
-  
+
   export interface Options {
     debug?: boolean;
     protocol?: string;
@@ -1460,7 +1460,7 @@ declare namespace Github {
     & Number
     & {
       reviewers?: string[];
-      team_reviewers?: Array;
+      team_reviewers?: string[];
     };
   export type PullRequestsDeleteReviewRequestParams =
     & Owner
@@ -1468,7 +1468,7 @@ declare namespace Github {
     & Number
     & {
       reviewers?: string[];
-      team_reviewers?: Array;
+      team_reviewers?: string[];
     };
   export type ReactionsGetForCommitCommentParams =
     & Owner

--- a/lib/index.d.ts
+++ b/lib/index.d.ts
@@ -15,7 +15,7 @@ declare namespace Github {
 
   export interface EmptyParams {
   }
-
+  
   export interface Options {
     debug?: boolean;
     protocol?: string;
@@ -1629,7 +1629,7 @@ declare namespace Github {
     & Owner
     & Repo
     & {
-      names: Array;
+      names: string[];
     };
   export type ReposGetContributorsParams =
     & Owner

--- a/lib/index.js.flow
+++ b/lib/index.js.flow
@@ -1455,7 +1455,7 @@ declare module "github" {
     & Number
     & {
       reviewers?: string[];
-      team_reviewers?: Array;
+      team_reviewers?: string[];
     };
   declare type PullRequestsDeleteReviewRequestParams =
     & Owner
@@ -1463,7 +1463,7 @@ declare module "github" {
     & Number
     & {
       reviewers?: string[];
-      team_reviewers?: Array;
+      team_reviewers?: string[];
     };
   declare type ReactionsGetForCommitCommentParams =
     & Owner
@@ -1624,7 +1624,7 @@ declare module "github" {
     & Owner
     & Repo
     & {
-      names: Array;
+      names: string[];
     };
   declare type ReposGetContributorsParams =
     & Owner

